staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware_test.go

```diff
@@ -496,9 +498,15 @@ func TestProxyUpgradeErrorResponse(t *testing.T) {
 		expectedErr = errors.New("EXPECTED")
 	)
 	proxy := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		transport := http.DefaultTransport.(*http.Transport)
-		transport.Dial = func(network, addr string) (net.Conn, error) {
-			return &fakeConn{err: expectedErr}, nil
+		transport := &http.Transport{
```

Comment:
don't we have a Transport copier somewhere? I believe we do.

Reply:
The closest thing I found is `utilnet.SetOldTransportDefaults()` which is used above. It does not matter, it is just a test. Here I just copied the definition of the default http transport. Transport contains a mutex so it is not safe to copy the whole struct.

Reply:
true, for the test it does not matter